supernode/daemon/mgr/task/manager_util.go

```diff
@@ -464,17 +475,21 @@ func isWait(CDNStatus string) bool {
 	return CDNStatus == types.TaskInfoCdnStatusWAITING
 }
 
-func getHTTPFileLength(taskID, url string, headers map[string]string) (int64, error) {
+func (tm *Manager) getHTTPFileLength(taskID, url string, headers map[string]string) (int64, error) {
 	fileLength, code, err := getContentLength(url, headers)
 	if err != nil {
-		return -1, err
+		return 0, errors.Wrapf(errorType.ErrUnknowError, "failed to get http file Length: %v", err)
 	}
 
 	if code == http.StatusUnauthorized || code == http.StatusProxyAuthRequired {
-		return -1, errors.Wrapf(errorType.ErrAuthenticationRequired, "taskID: %s,code: %d", taskID, code)
+		return 0, errors.Wrapf(errorType.ErrAuthenticationRequired, "taskID: %s,code: %d", taskID, code)
 	}
 	if code != http.StatusOK {
 		logrus.Warnf("failed to get http file length with unexpected code: %d", code)
+		if code == http.StatusNotFound {
+			tm.taskURLUnReachableStore.Add(taskID, time.Now())
```

Comment:
Please do not make `getHTTPFileLength` to be a function of `*Manager`, because of adding this line of code. It works, but unreasonable for readability and unit testing.

Reply:
Done.